src/boot/efi/proto-file-io.h

```diff
@@ -0,0 +1,77 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include "efi.h"
+
+#define EFI_SIMPLE_FILE_SYSTEM_PROTOCOL_GUID \
+        GUID_DEF(0x0964e5b22, 0x6459, 0x11d2, 0x8e, 0x39, 0x00, 0xa0, 0xc9, 0x69, 0x72, 0x3b)
+#define EFI_FILE_INFO_ID \
+        GUID_DEF(0x009576e92, 0x6d3f, 0x11d2, 0x8e, 0x39, 0x00, 0xa0, 0xc9, 0x69, 0x72, 0x3b)
+
+#define EFI_FILE_MODE_READ   0x0000000000000001
+#define EFI_FILE_MODE_WRITE  0x0000000000000002
+#define EFI_FILE_MODE_CREATE 0x8000000000000000
```

Comment:
so many zeroes....

I'd really make this all use UINT32_C() or suffix U because otherwise:

```c
typeof(EFI_FILE_MODE_READ) x = EFI_FILE_MODE_READ;
typeof(EFI_FILE_MODE_CREATE) y = EFI_FILE_MODE_CREATE;

assert(x < y);
```

would have an undefined result, since you'd compare a signed with an unsigned value, and `y` would change its sign if it would ne converted to int